acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/dsl/Contracts.java

```diff
@@ -32,15 +37,33 @@
   public static final BigInteger GAS_LIMIT = BigInteger.valueOf(3000000);
 
   private final Eth eth;
+  private final Eea eea;
 
-  public Contracts(final Eth eth) {
+  public Contracts(final Eth eth, final Eea eea) {
     this.eth = eth;
+    this.eea = eea;
   }
 
   public String submit(final Transaction smartContract) {
     return failOnIOException(() -> eth.sendTransaction(smartContract));
   }
 
+  public String submitPrivateTransaction(final PrivateTransaction smartContract) {
```

Comment:
Private transactions are a different specialisation (as `code()` isn't going to work. 

I'd say that this contract class is a prime case for applying inheritance, one for public contracts and another for private contracts 😉

Reply:
done. split into publicContracts and privateContracts